staging/src/k8s.io/legacy-cloud-providers/azure/azure_controller_standard.go

```diff
@@ -117,8 +116,7 @@ func (as *availabilitySet) DetachDisk(diskName, diskURI string, nodeName types.N
 		return nil, err
 	}
 
-	disks := make([]compute.DataDisk, len(*vm.StorageProfile.DataDisks))
-	copy(disks, *vm.StorageProfile.DataDisks)
+	disks := filterDetachingDisks(*vm.StorageProfile.DataDisks)
```

Comment:
same here, check whether the disk is under detaching and fail earlier

Reply:
no need to check